pennylane/gradients/parameter_shift_cv.py

```diff
@@ -0,0 +1,678 @@
+                "gradient recipe of more than two terms."
+            )
+
+        shifted_tapes = generate_shifted_tapes(tape, idx, shifts, multipliers)
+
+        # evaluate transformed observables at the original parameter point
+        # first build the Heisenberg picture transformation matrix Z
+        Z0 = op.heisenberg_tr(dev_wires, inverse=True)
+        Z2 = shifted_tapes[0]._par_info[t_idx]["op"].heisenberg_tr(dev_wires)
+        Z1 = shifted_tapes[1]._par_info[t_idx]["op"].heisenberg_tr(dev_wires)
+
+        # derivative of the operation
+        Z = Z2 * coeffs[0] + Z1 * coeffs[1]
+        Z = Z @ Z0
+
+        # conjugate Z with all the descendant operations
+        B = np.eye(1 + 2 * len(dev_wires))
+        B_inv = B.copy()
+
+        succ = tape.graph.descendants_in_order((op,))
+        operation_descendents = itertools.filterfalse(qml.circuit_graph._is_observable, succ)
+        observable_descendents = filter(qml.circuit_graph._is_observable, succ)
+
+        for BB in operation_descendents:
+            if not BB.supports_heisenberg:
+                # if the descendant gate is non-Gaussian in parameter-shift differentiation
+                # mode, then there must be no observable following it.
+                continue
+
+            B = BB.heisenberg_tr(dev_wires) @ B
+            B_inv = B_inv @ BB.heisenberg_tr(dev_wires, inverse=True)
+
+        Z = B @ Z @ B_inv  # conjugation
+
+        g_tape = tape.copy(copy_operations=True)
+        constants = []
+
+        # transform the descendant observables into their derivatives using Z
+        transformed_obs_idx = []
+
+        for obs in observable_descendents:
+            # get the index of the descendent observable
+            idx = tape.observables.index(obs)
+            transformed_obs_idx.append(idx)
+
+            transformed_obs = _transform_observable(obs, Z, dev_wires)
+
+            A = transformed_obs.parameters[0]
+            constants.append(None)
+
+            # Check if the transformed observable corresponds to a constant term.
+            if len(A.nonzero()[0]) == 1:
+                if A.ndim == 2 and A[0, 0] != 0:
+                    constants[-1] = A[0, 0]
+
+                elif A.ndim == 1 and A[0] != 0:
+                    constants[-1] = A[0]
+
+            g_tape._measurements[idx] = qml.measure.MeasurementProcess(
+                qml.operation.Expectation, _transform_observable(obs, Z, dev_wires)
+            )
+
+        if not any(i is None for i in constants):
+            # Check if *all* transformed observables corresponds to a constant term.
+            # term. If this is the case for all transformed observables on the tape,
+            # then <psi|A|psi> = A<psi|psi> = A,
+            # and we can avoid the device execution.
+            shapes[-1] = 0
+            obs_indices.append(transformed_obs_idx)
+            gradient_values.append(constants)
+            continue

```

Comment:
This part is new, and corresponds to this TODO: https://github.com/PennyLaneAI/pennylane/blob/ab71001bc55c2945a0945f42131c45d77da3512e/pennylane/tape/cv_param_shift.py#L344